Assets/Scripts/Logic/AIPlayer.cs

```diff
@@ -0,0 +1,127 @@
+
+        public Tile Tile 
+        {
+            get => _tile;
+            set
+            {
+                if (_tile != null)
+                {
+                    _tile.IsOccupied = false;
+                }
+                _tile = value;
+                _tile.IsOccupied = true;
+            }
+        }
+        
+        private int _stepsCount;
+        private int _lastAttackStep;
+        private int _stepsInCurrentAttack;
+        public int Id { get; set; }
+        public int Health { get;  set; }
+
+        private AIPlayer() { }
+        public AIPlayer(int movementsPerStep, int stepsPerAttack,
+            int attackRange, int health, Tile startingTile, IBoard board)
+        {
+            Id = _nextId++;
+            _movementsPerStep = movementsPerStep;
+            _stepsPerAttack = stepsPerAttack;
+            _attackRange = attackRange;
+            Health = health;
+            Tile = startingTile;
+            _board = board;
+            
+            PlayerDied += () => Debug.Log($"Player {Id} died");
+        }
+
+        public void TakeDamage(int damage)
+        {
+            if (!IsAlive)
+                return;
+            Health -= damage;
+            if(!IsAlive)
+                PlayerDied?.Invoke();
+        }
+
+        public IReadOnlyList<ICommand> Step()
+        {
+            var commands = new List<ICommand>();
+            
+            if (!IsAlive)
+                return commands;
+            
+            _stepsCount++;
+            var opponent = PlayerManager.Instance.GetOpponent(this);
+            var distance = Tile.Position.GetDistance(opponent.Tile.Position);
+            
+            for(int i = 0; i < _movementsPerStep; i++)
+            {
+                if (distance <= _attackRange)
+                {
+                    commands.Add(Attack(opponent));
+                    return commands;
+                }
+                
+                commands.Add(MoveTowards(opponent));
+            }
+
+            return commands;
+        }
+
+        private AttackCommand Attack(IPlayer opponent)
+        {
+            Debug.Log($"Player {ToString()} " +
+                      $"attacks Player {opponent}. " +
+                      $"Distance is {Tile.Position.GetDistance(opponent.Tile.Position)}");
+            
+            if (_stepsCount - _lastAttackStep != 1)
+            {
+                _stepsInCurrentAttack = 0;
+            }
+            
+            _stepsInCurrentAttack++;
+            
+            if (_stepsInCurrentAttack == _stepsPerAttack)
+            {
+                _stepsInCurrentAttack = 0;
+                opponent.TakeDamage(1);
+            }
+            
+            _lastAttackStep = _stepsCount;
+            return new AttackCommand();
+        }
+
+        private MoveCommand MoveTowards(IPlayer opponent)
+        {
+            var position = opponent.Tile.Position;
+            var nextTile = Tile.GetTileTowards(position);
```

Comment:
`Tile.GetTileTowards(position)` should guarantee that a non-null tile will be returned. Otherwise, it can also break the `ToString()` method.